hl13/code/modules/projectiles/ammunition/ballistic/pistol.dm

```diff
@@ -3,3 +3,9 @@
 	desc = "A 9mm bullet casing."
 	caliber = CALIBER_9MM
 	projectile_type = /obj/projectile/bullet/hl13/hl9mm
+
+/obj/item/ammo_casing/hl13/hl9mm/improvised
```

Comment:
did we really need an improvised bullet if we already have the standard bullets? I mean, even irl making a bullet isnt hard (you can even fill empty shells) so i think you should skip improvised bullet and make a way to recover used casings and refill them or just find the material to craft them (recipes are supposed to be locked if a guy doesnt have the knowledge on how to craft stuff anyway)

Reply:
If you want to keep it anyway, it needs to be lower quality so it should either fail to shoot some times or make your gun jam.. Also it needs to be distinguishable from the other standard button so it needs to have different sprite for basically everything

Reply:
Did the empty shell filling

Reply:
issue still present